crates/cheatcodes/src/script.rs

```diff
@@ -33,63 +33,91 @@ impl Cheatcode for broadcast_2Call {
-            address: *implementation,
-            nonce: *nonce,
-            chain_id: U256::from(ccx.ecx.env.cfg.chain_id),
-        };
-        let signed_auth = SignedAuthorization::new_unchecked(
-            auth,
-            *v,
-            U256::from_be_bytes(r.0),
-            U256::from_be_bytes(s.0),
-        );
-        write_delegation(ccx, signed_auth.clone())?;
-        ccx.state.active_delegation = Some(signed_auth);
-        Ok(Default::default())
+impl Cheatcode for attachDelegation_1Call {
+    fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
+        let Self { signedDelegation, crossChain } = self;
+        attach_delegation(ccx, signedDelegation, *crossChain)
     }
 }
 
 impl Cheatcode for signDelegation_0Call {
     fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
         let Self { implementation, privateKey } = *self;
-        sign_delegation(ccx, privateKey, implementation, None, false)
+        sign_delegation(ccx, privateKey, implementation, None, false, false)
     }
 }
 
 impl Cheatcode for signDelegation_1Call {
     fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
         let Self { implementation, privateKey, nonce } = *self;
-        sign_delegation(ccx, privateKey, implementation, Some(nonce), false)
+        sign_delegation(ccx, privateKey, implementation, Some(nonce), false, false)
+    }
+}
+
+impl Cheatcode for signDelegation_2Call {
+    fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
+        let Self { implementation, privateKey, crossChain } = *self;
+        sign_delegation(ccx, privateKey, implementation, None, crossChain, false)
     }
 }
 
 impl Cheatcode for signAndAttachDelegation_0Call {
     fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
         let Self { implementation, privateKey } = *self;
-        sign_delegation(ccx, privateKey, implementation, None, true)
+        sign_delegation(ccx, privateKey, implementation, None, false, true)
     }
 }
 
 impl Cheatcode for signAndAttachDelegation_1Call {
     fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
         let Self { implementation, privateKey, nonce } = *self;
-        sign_delegation(ccx, privateKey, implementation, Some(nonce), true)
+        sign_delegation(ccx, privateKey, implementation, Some(nonce), false, true)
     }
 }
 
+impl Cheatcode for signAndAttachDelegation_2Call {
+    fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
+        let Self { implementation, privateKey, crossChain } = *self;
+        sign_delegation(ccx, privateKey, implementation, None, crossChain, true)
+    }
+}
+
+/// Helper function to attach an EIP-7702 delegation.
+fn attach_delegation(
+    ccx: &mut CheatsCtxt,
+    delegation: &SignedDelegation,
+    cross_chain: bool,
+) -> Result {
+    let SignedDelegation { v, r, s, nonce, implementation } = delegation;
+    let chain_id = if cross_chain { U256::from(0) } else { U256::from(ccx.ecx.env.cfg.chain_id) };
```

Comment:
Would you mind adding a comment on how chain 0 behavior enables cross-chain validity here?

Ref: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7702.md#protection-from-malleability-cross-chain

Reply:
added https://github.com/foundry-rs/foundry/pull/10518/commits/a5a617638d9ca597a36d8fe6a9ffedd1fcb690e2 hope makes sense